packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -195,7 +195,15 @@ class DatePickerInput extends React.Component {
 
     const label = this.props.ariaLabel ? this.props.ariaLabel : intl.formatMessage({ id: 'Terra.datePicker.date' });
 
-    this.formatDescriptionId = (useExternalFormatMask === false) ? `terra-date-picker-description-format-${this.uuid}` : '';
+    if (useExternalFormatMask === false && inputAttributes && inputAttributes['aria-describedby']) {
+      this.formatDescriptionId = `terra-date-picker-description-format-${this.uuid} ${inputAttributes['aria-describedby']}`;
```

Comment:
Would this add an empty space between the uuid and aria-describedby?

Reply:
I think the issue from [this comment](https://github.com/cerner/terra-framework/pull/1252#discussion_r492108887) still exists with this new change.  `useExternalFormatMask` defaults to false for DatePicker and if the consumer provides `aria-describedby`, we are prepending a uuid to its aria-describedby and pass that down to Input. Consumers probably wouldn't expect their `aria-describedby` to get updated.

Or maybe I'm not understanding what this is trying to do.

Reply:
just curious, is the top most if block the most likely scenario? it's ideal if the conditionals follow most likely to least likely :D

Reply:
> I think the issue from this comment still exists with this new change. useExternalFormatMask defaults to false for DatePicker and if the consumer provides aria-describedby, we are prepending a uuid to its aria-describedby and pass that down to Input. Consumers probably wouldn't expect their aria-describedby to get updated.
Or maybe I'm not understanding what this is trying to do.

So, earlier we didn't have any description id's being passed down to the raw component from the field component due to `inputAtrributes`.
Correct me if I am wrong but I believe that we should read the `format, error and help` text, and then any of the user sent `aria-describedby`.


cc / @neilpfeiffer

Reply:
>just curious, is the top most if block the most likely scenario? it's ideal if the conditionals follow most likely to least likely :D

If we want it in if-else order then I think this order of condition is the best. But we can also provide single if blocks, and take care of preference there.

Reply:
>Would this add an empty space between the uuid and aria-describedby?

Yes, it would add spaces between them only when `aria-describedby` is provided.

And for readability I can change the name of `this.formatDescriptionId` to `this.ariaDescriptionId`.

What do you think? @benbcai

Reply:
> > Would this add an empty space between the uuid and aria-describedby?
> 
> Yes, it would add spaces between them only when `aria-describedby` is provided.
> 
> And for readability I can change the name of `this.formatDescriptionId` to `this.ariaDescriptionId`.
> 
> What do you think? @benbcai

I just think it's weird that the `id` as a string has two separate strings.

Reply:
Updated here https://github.com/cerner/terra-framework/pull/1252/commits/b10e52f5589813e424d2e9a198e285884d7eaef4